panic when using --strict-host-key-checking=false

```
fleetctl --strict-host-key-checking=false ssh 0357eef5-293a-4203-a406-771f9260e8d6 uptime
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x0 pc=0x4f6690]

goroutine 10 [running]:
runtime.panic(0x736d00, 0xbe2628)
    /usr/local/go/src/pkg/runtime/panic.c:266 +0xb6
github.com/coreos/fleet/ssh.(*HostKeyChecker).Check(0x0, 0xc210087bd0, 0xf, 0x7fb77a74ca18, 0xc210095cc0, ...)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/ssh/known_hosts.go:75 +0x130
github.com/coreos/fleet/ssh.*HostKeyChecker.Check·fm(0xc210087bd0, 0xf, 0x7fb77a74ca18, 0xc210095cc0, 0x7fb77a752548, ...)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/ssh/ssh.go:149 +0x7f
github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh.(*handshakeTransport).client(0xc21005d0f0, 0x7fb77a74c190, 0xc210036450, 0xc2100a2180, 0xc210058de0, ...)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh/handshake.go:391 +0x1e2
github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh.(*handshakeTransport).enterKeyExchange(0xc21005d0f0, 0xc2100a4400, 0x37d, 0x37d, 0x0, ...)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh/handshake.go:344 +0x706
github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh.(*handshakeTransport).readOnePacket(0xc21005d0f0, 0xc21005b270, 0x69a7c0, 0xc21009d110, 0x146, ...)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh/handshake.go:171 +0x158
github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh.(*handshakeTransport).readLoop(0xc21005d0f0)
    /home/core/fleet/gopath/src/github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh/handshake.go:138 +0x27
created by github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh.newClientTransport
    /home/core/fleet/gopath/src/github.com/coreos/fleet/third_party/code.google.com/p/gosshnew/ssh/handshake.go:101 +0xee
```

--- a/ssh/ssh.go
+++ b/ssh/ssh.go
@@ -146,7 +146,10 @@ func sshClientConfig(user string, checker *HostKeyChecker) (*gossh.ClientConfig,
 		Auth: []gossh.AuthMethod{
 			gossh.PublicKeys(signers...),
 		},
-		HostKeyCallback: checker.Check,
+	}
+
+	if checker != nil {
+		cfg.HostKeyCallback = checker.Check
 	}
 
 	return &cfg, nil

